### ISSUE TITLE
DebugSymbol.fromAddress  crash on iOS with version 14.1.2 gadget

### ISSUE THREAD

**Issue by zhangbozhb** [frida/frida#1561]
when i parse symbols with DebugSymbol.fromAddress in Interceptor.attach -> onEnter(args)  functions.
env: Mac os big Sur + iOS 14.2 + Xcode 12.2 + python 3.8.2 + node 15.3 + Frida 14.1.2

other issues:
hang up: iOS gadget 14.1.2 
When I use Interceptor.attach to some functions at launch time of app(mode: resume), It hang up the whole app, and killed by iOS system watchdog.  
case 1:  Interceptor.attach to nsapplication delegates at launch time;
case 2: access some global instance(NSNotificationCenter.defaultCenter) at launch time. am I wrong usage of api?

**Comment by 0x3c3e** [comment]
@zhangbozhb it seems broken since 14.1.x

**Comment by oleavr** [comment]
@zhangbozhb Thanks for reporting! Is the device jailbroken? Are you repackaging the app yourself or using Frida's built-in jailed iOS support?

**Comment by zhangbozhb** [comment]
@oleavr No. Normal device without jailbroken.  I just inject my app with gadget.dylib and repacking it with development profile, then install on iPhone 11 with OS 14.2; I use python to talk to app, and run javascript script written with  javascript apis.
Beside, if  the app(injected with gadget) has launch, then I attach to that app on iPhone device, and everything go well. After those steps, run codes as follows
`
 script = self.session.create_script(frida_code)
script.set_log_handler(self.log_handler)
script.on('message', self.handle_message_from_js)
 script.load()
`
it fail with error "v8 runtime not available due to build configuration"

**Comment by zhangbozhb** [comment]
@oleavr Is there any update on crash issue.

**Comment by zhangbozhb** [comment]
I found if i do not inject gadget.dylib, it works fine on ios 14.

**Comment by oleavr** [comment]
May have fixed this yesterday:
https://github.com/frida/frida-core/commit/fc7b554cc873e82f5ee93d11578ac2cc2aca5436
Could you give it a try? (Build from git but make sure the frida-core submodule is at latest master as I didn't yet update the submodules in the `frida` repo.)

**Comment by zhangbozhb** [comment]
> May have fixed this yesterday:
> [frida/frida-core@fc7b554](https://github.com/frida/frida-core/commit/fc7b554cc873e82f5ee93d11578ac2cc2aca5436)
> Could you give it a try? (Build from git but make sure the frida-core submodule is at latest master as I didn't yet update the submodules in the `frida` repo.)

I use latest version with Frida 14.2.13 and run iOS 14.4 with iPhone 11. I'm sure my device is not jailbreak.
I do nothing but install app to iphone with xcode,  frida works well.
Howerver if I manully modify ipa and inject gadget.dylib and then install, frida always crash.

I am curious how gadgets are automatically injected into the app, and my iPhone is not jailbreak.
I notice that from doc "Frida is able to instrument debuggable apps, and will inject Gadget automatically as of Frida 12.7.12."
If jailbreak is not required, and it can inject Gadget.dylib. Similarly, can we inject other dylibs in the same way? If the answer is yes, how and what should we do to achieve this goal. In addition, is there any documents or materials in this field. I failed to find related code and document in frida code.
